src/afrolabs/components/kafka.clj

```diff
@@ -1285,9 +1316,14 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 (defn assert-topics
+  "Makes sure topics exist with the given number of partitions.
+  If a topic exists and has fewer partitions than specified with `nr-of-partitions`, the partitions will be
+  increased to match (when `increase-existing-partitions?` is `true`)."
   [^AdminClient admin-client
    desired-topics
-   & {:keys [nr-of-partitions]}]
+   & {:keys [nr-of-partitions
+             increase-existing-partitions?]
+      :or   {increase-existing-partitions? false}}]
```

Comment:
partitions may grow to the specified `nr-of-partitions`, but this behaviour is governed by a flag to opt-in.